lib/internal/worker.js

```diff
@@ -57,6 +59,17 @@ function debug(...args) {
   debuglog(...args);
 }
 
+let cwdCounter;
+
+if (isMainThread) {
+  cwdCounter = new Uint32Array(new SharedArrayBuffer(4));
+  const originalCwd = process.chdir;
+  process.chdir = function(cwd) {
+    Atomics.add(cwdCounter, 0, 1);
+    originalCwd(cwd);
```

Comment:
Correct me if I'm wrong but isn't there a race window here? Consider the following sequence of events:

1. main increments the counter
2. worker observes the incremented counter
3. worker fetches the cwd which hasn't changed yet so it gets the old cwd
4. main chdirs
5. worker now has an outdated cwd

There needs to be a critical section where worker threads block or spin until the main thread is done chdir'ing.

Reply:
I think this is race-y anyway, because concurrent `process.chdir()` and `process.cwd()` without external synchronization are inherently?

What if the counter is increased after the modification? I don’t think that leads to a situation that can’t already happen.